### ISSUE TITLE
Won't work together with the python wrapper

### ISSUE THREAD

**Issue by yuriy-davygora** [conda-forge/kyotocabinet-feedstock#1]
Issue: Won't work together with the python wrapper

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list

# packages in environment at /home/yuriyd/miniconda3/envs/test:
#
# Name                    Version                   Build  Channel
bzip2                     1.0.6                h470a237_2    conda-forge
ca-certificates           2018.4.16                     0    conda-forge
certifi                   2018.8.13                py27_0    conda-forge
kyotocabinet              1.2.76               hd28b015_0    conda-forge
kyotocabinet              1.10                      <pip>
libgcc-ng                 7.2.0                hdf63c60_3  
libstdcxx-ng              7.2.0                hdf63c60_3  
ncurses                   6.1                  hfc679d8_1    conda-forge
openssl                   1.0.2o               h470a237_1    conda-forge
pip                       18.0                     py27_1    conda-forge
python                    2.7.15               h9fef7bc_0    conda-forge
readline                  7.0                  haf1bffa_1    conda-forge
setuptools                40.0.0                   py27_1    conda-forge
sqlite                    3.24.0               h2f33b56_0    conda-forge
tk                        8.6.8                         0    conda-forge
wheel                     0.31.1                   py27_1    conda-forge
zlib                      1.2.11               h470a237_3    conda-forge

```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

     active environment : test
    active env location : /home/yuriyd/miniconda3/envs/test
            shell level : 1
       user config file : /home/yuriyd/.condarc
 populated config files : /home/yuriyd/.condarc
          conda version : 4.5.10
    conda-build version : 3.12.1
         python version : 3.6.5.final.0
       base environment : /home/yuriyd/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/linux-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /home/yuriyd/miniconda3/pkgs
                          /home/yuriyd/.conda/pkgs
       envs directories : /home/yuriyd/miniconda3/envs
                          /home/yuriyd/.conda/envs
               platform : linux-64
             user-agent : conda/4.5.10 requests/2.18.4 CPython/3.6.5 Linux/4.15.0-30-generic ubuntu/18.04 glibc/2.27
                UID:GID : 1001:1001
             netrc file : None
           offline mode : False

```
</details>

Steps to reproduce the problem

<details>

1. conda create -n test python=2.7.15
2. conda activate test
3. conda install kyotocabinet
4. pip install kyotocabinet
5. echo "import kyotocabinet" | python

</details>

Error message:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: /home/yuriyd/miniconda3/envs/test/lib/python2.7/site-packages/kyotocabinet.so: undefined symbol: _ZN12kyotocabinet9DirStream4openERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE

```

Additional info:

<details>

If I build the conda-forge package locally (from https://github.com/yuriy-davygora/staged-recipes/tree/kyotocabinet) and follow the above described steps, but installing the locally build conda package instead of the one from conda-forge, the import statement causes no error.

</details>

<br/>

Any help will be greatly appreciated

**Comment by yuriy-davygora** [comment]
More info here: My default system compiler was gcc-7.3.0, and, apparently, the library has been built with gcc-4.*. After I switched the system compiler to gcc-4.8.0, cleared my pip cache and reinstalled the environment, everything worked.

I suppose, lack of backward compatibility of g++ is not something that has to be dealt with within this scope, so I am closing the issue.

**Comment by djsutherland** [comment]
Yeah, gcc7 versus 4 has ABI compatibility issues. You might be able to compile the python wrapper adding `-D_GLIBCXX_USE_CXX11_ABI=0` to `CXXFLAGS` or something and get it to work with gcc7; or, you could put an issue on [staged-recipes](https://github.com/conda-forge/staged-recipes/) requesting us to package the Python wrapper in conda-forge anyway, which would also work. :)

We're in the process of migrating all of conda-forge to using the new c++ ABI, but it's a slow process.

**Comment by yuriy-davygora** [comment]
@dougalsutherland Thank you for your comment and for your suggestions. Since it was the problem only with my machine and since we have GCC 4 on our cluster machines, this is not a big issue. Everything works quite fine.

Speaking of the python wrapper, though: we have found some problems there, which we are currently looking at. We might actually have to patch it slightly, and then, depending on whether or not the original maintainer is still active, either submit it to the upstream, or create our own fork. I will get back to you then, when we have finished our investigations.
